rapids-4-spark-tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala

```diff
@@ -229,23 +342,142 @@ class ApplicationInfo(
+    } else {
+      logError("No Job Found. Exiting.")
+      System.exit(1)
+    }
+
+    // For stageDF
+    if (stageSubmitted.nonEmpty) {
+      val stageSubmittedNew: ArrayBuffer[StageCase] = ArrayBuffer[StageCase]()
+      for (res <- stageSubmitted) {
+        val thisEndTime = stageCompletionTime(res.stageId)
+        val thisFailureReason = stageFailureReason(res.stageId)
+
+        val durationResult = ProfileUtils.optionLongMinusOptionLong(thisEndTime, res.submissionTime)
+        val durationString = durationResult match {
+          case Some(i) => UIUtils.formatDuration(i)
+          case None => ""
+        }
+
+        val stageNew = res.copy(completionTime = thisEndTime,
+          failureReason = thisFailureReason,
+          duration = durationResult,
+          durationStr = durationString)
+        stageSubmittedNew += stageNew
+      }
+      allDataFrames += (s"stageDF_$index" -> stageSubmittedNew.toDF)
+    } else {
+      logError("No Stage Found. Exiting.")
+      System.exit(1)
+    }
+
+    // For taskDF
+    if (taskEnd.nonEmpty) {
+      allDataFrames += (s"taskDF_$index" -> taskEnd.toDF)
+    } else {
+      logError("task is empty! Exiting...")
+      System.exit(1)
+    }
+
+    // For sqlMetricsDF
+    if (sqlPlanMetrics.nonEmpty) {
+      logInfo(s"Total ${sqlPlanMetrics.size} SQL Metrics for appID=$appId")
+      allDataFrames += (s"sqlMetricsDF_$index" -> sqlPlanMetrics.toDF)
+    } else {
+      logInfo("No SQL Metrics Found. Skipping generating SQL Metrics DataFrame.")
+    }
+
+    // For driverAccumDF
+    allDataFrames += (s"driverAccumDF_$index" -> driverAccum.toDF)
+    if (driverAccum.nonEmpty) {
+      logInfo(s"Total ${driverAccum.size} driver accums for appID=$appId")
+    } else {
+      logInfo("No Driver accum Found. Create an empty driver accum DataFrame.")
+    }
+
+    // For taskStageAccumDF
+    allDataFrames += (s"taskStageAccumDF_$index" -> taskStageAccum.toDF)
+    if (taskStageAccum.nonEmpty) {
+      logInfo(s"Total ${taskStageAccum.size} task&stage accums for appID=$appId")
+    } else {
+      logInfo("No task&stage accums Found.Create an empty task&stage accum DataFrame.")
+    }
+
+    // For planNodeAccumDF
+    allDataFrames += (s"planNodeAccumDF_$index" -> planNodeAccum.toDF)
+    if (planNodeAccum.nonEmpty) {
+      logInfo(s"Total ${planNodeAccum.size} Plan node accums for appID=$appId")
+    } else {
+      logInfo("No Plan node accums Found. Create an empty Plan node accums DataFrame.")
+    }
+
+    // For problematicSQLDF
+    allDataFrames += (s"problematicSQLDF_$index" -> problematicSQL.toDF)
+
     for ((name, df) <- this.allDataFrames) {
       df.createOrReplaceTempView(name)
       sparkSession.table(name).cache
     }
   }
 
-  // Function to run a query and show the result -- limit 1000 rows.
-  def runQuery(query: String, vertical: Boolean = false): DataFrame = {
+  // Function to run a query and print the result to the file -- limit 1000 rows.
```

Comment:
why limit 1000 here? we want all rows to go to file, remove or possibly make configurable.

Reply:
I have made it configurable.